src/app/clusters/boolean-state-configuration-server/boolean-state-configuration-server.h

```diff
@@ -0,0 +1,56 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file API declarations for boolean state configuration cluster.
+ */
+
+#pragma once
+
+#include "boolean-state-configuration-delegate.h"
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/util/basic-types.h>
+#include <lib/core/DataModelTypes.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace BooleanStateConfiguration {
+
+void SetDefaultDelegate(EndpointId endpoint, Delegate * delegate);
+Delegate * GetDefaultDelegate(EndpointId endpoint);
+
+CHIP_ERROR SetAlarmsActive(EndpointId ep, chip::BitMask<AlarmModeBitmap> alarms);
+CHIP_ERROR SetAllEnabledAlarmsActive(EndpointId ep);
+CHIP_ERROR ClearAllAlarms(EndpointId ep);
+CHIP_ERROR SuppressAlarms(EndpointId ep, chip::BitMask<BooleanStateConfiguration::AlarmModeBitmap> alarms);
+CHIP_ERROR SetCurrentSensitivityLevel(EndpointId ep, uint8_t level);
+CHIP_ERROR EmitSensorFault(EndpointId ep, chip::BitMask<BooleanStateConfiguration::SensorFaultBitmap> fault);
+
```

Comment:
As a **follow-up** (not this PR): please move these to be method of an instance, rather than global functions.

Reply:
An instance would make sense if this was a single cluster instance on a node. But I think this cluster can exist on multiple endpoints?

Reply:
I've created https://github.com/project-chip/connectedhomeip/issues/31428 and I will mark this conversation as resolved. Please let me know your feedback on my previous comment.